src/arfsdao.ts

```diff
@@ -850,9 +847,6 @@ export class ArFSDAO extends ArFSDAOAnonymous {
 		maxDepth: number,
 		includeRoot: boolean
 	): Promise<ArFSPrivateFileOrFolderWithPaths[]> {
-		if (maxDepth < 0 || !Number.isInteger(maxDepth)) {
```

Comment:
Why remove these guards? If you had left them in, you'd have seen that using POSITIVE_INFINITY doesn't actually work 🤔